utils/Span.h

```diff
@@ -0,0 +1,90 @@
+#pragma once
+
+#include <type_traits>
+
+namespace utils
+{
+
+/**
+ * The class template span describes an object that can refer to a contiguous sequence of objects with the first element
+ * of the sequence at position zero.
+ *
+ * This is intends to be an implementation of std::span as it only available on C++20.
+ */
+template <class T>
+class Span
```

Comment:
Quite useful to turn raw vectors into iteratable container. We probably have several places where we can apply it in iterations.

Reply:
Yes and I see other situations where this can be useful too. Like when we want want give just a view of a subpart a vector/array. And to have a similar interface regardless what the container is so we can use utils::Span instead of using templates to receive parameters that can be Carrays, std::arrays or std::vectors